include/circt/Dialect/Handshake/HandshakeInterfaces.td

```diff
@@ -15,6 +15,15 @@
 
 include "mlir/IR/OpBase.td"
 
+def FineGrainedDataflowRegionOpInterface : OpInterface<"FineGrainedDataflowRegionOpInterface"> {
```

Comment:
Since this doesn't have any methods, could this be a Trait? I'm not sure if that helps at all with your HasParent woes, but if this is just a marker with no methods, perhaps a Trait is more appropriate?

Reply:
Could be a trait, yea, but just shuffles the problem over to something like `ParentHasTrait`, which doesn't exist either. Personally i prefer this pattern to be checking an `OpInterface` seeing as, while this doesn't have any methods _yet_, i could easily imagine that it would have in a later stage.